src/olympia/accounts/views.py

```diff
@@ -197,9 +198,15 @@ def inner(self, request):
                     request, ERROR_STATE_MISMATCH, next_path=next_path,
                     format=format)
             elif request.user.is_authenticated():
-                return render_error(
+                response = render_error(
                     request, ERROR_AUTHENTICATED, next_path=next_path,
                     format=format)
+                # If the api token cookie is missing but we're still
+                # authenticated using the session, add it back.
+                if API_TOKEN_COOKIE not in request.COOKIES:
```

Comment:
Can you add some logging to say that an API token was regenerated and the cookie was reset?